crates/core/src/binding.rs

```diff
@@ -506,9 +507,18 @@ impl<'a> Binding<'a> {
             Binding::Node(source, _) => Some(source),
             Binding::String(source, _) => Some(source),
             Binding::List(source, _, _) => Some(source),
-            // maybe should be none?
-            Binding::FileName(source) => Some(source),
-            Binding::ConstantRef(_) => None,
+            Binding::FileName(..) | Binding::ConstantRef(..) => None,
```

Comment:
Returning `None` (as the comment suggested) would've previously caused potential issues in `step.rs` where the source was compared against a filename. With the introduction of `as_filename`, this should no longer be an issue. At least the tests still pass :)